### PR TITLE
Introduce a global way to get APIs key easily

### PR DESCRIPTION
**Pull request by LuisAlbertoPenaNunez**
### This issue is related to #431 

# What's New

- I've been receiving feedback that the repository must be clean of any API key, and it shall be CI responsability to put them.

I included a new file called **MobileConfig.json** into the **Assets** folder, if you are going to add a new API key, just make a new property in the file

screen shot 2016-04-15 at 15 47 47

then go to the **MobileConfig** in **Core/MobileConfig** and add the new property so the JSON parser doesn't blow up.

screen shot 2016-04-15 at 15 24 43

Then to get the API keys, all you have to do is

**ViewModel**

screen shot 2016-04-15 at 15 27 15

Inject **IMobileConfigService** as a dependency, and then access the **MobileConfigurationFile** and there you have it.

### CODE REVIEW

File: Mobile/Core/IMobileConfigService.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Core
+{
+	public interface IMobileConfigService
+	{
+		MobileConfig MobileConfigurationFile { get; set; }
```
Comment (Willjobs94) [inline Mobile/Core/IMobileConfigService.cs]: What about rename it to MobileConfigurationManager ?

File: Mobile/Core/MobileConfig.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Core
+{
+	public class MobileConfig
+	{
+		public string GitHubKey { get; set; }
+
+		public string XamarinInsightKey { get; set; }
+
+		public string GooglePlacesAPI { get; set; }
```
Comment (Willjobs94) [inline Mobile/Core/MobileConfig.cs]: Just for convention this one should have "Key" at the end

File: Mobile/Core/IAssetReader.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.IO;
+using Android;
+
+namespace Core
+{
+	public interface IAssetReader
+	{
+		string ReadFileAsString(string fileName);
```
Comment (Willjobs94) [inline Mobile/Core/IAssetReader.cs]: Don't you need to have a path argument ? So we can inject it when working on iOS or it used the same path for both OS ?

File: Mobile/Core/MobileConfig.cs (hunk 2 above)
Comment (codnee) [inline Mobile/Core/MobileConfig.cs]: Yes, it should be GooglePlacesKey.

File: Mobile/Core/IMobileConfigService.cs (hunk 1 above)
Comment (LuisAlbertoPenaNunez) [inline Mobile/Core/IMobileConfigService.cs]: Bien

File: Mobile/Core/MobileConfig.cs (hunk 2 above)
Comment (LuisAlbertoPenaNunez) [inline Mobile/Core/MobileConfig.cs]: Era un ejemplo, **GooglePlacesAPI** va en el **Manifest** lamentablemente, pero hare un **rebase** anyways.

File: Mobile/Core/IAssetReader.cs (hunk 3 above)
Comment (LuisAlbertoPenaNunez) [inline Mobile/Core/IAssetReader.cs]: 1) Estamos trabajando con plataformas **mobiles**, no podemos decir que proveen un **Path**, por que cada cual tiene su forma de manejar **"Assets extra"**.

2) Cada plataforma proveera el **Base Path**, por ejemplo en **Android** solo puedes tener "cualquier archivo" en el folder **Assets**.  :)

**Pull request merged**